### PR TITLE
Feature send context time to committer

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

**Comment by 1p4pk** [comment]
@raminqaf How about starting with a new branch Ramin? So we do not trash PRs. BEsides, spotless does not work.

**Review by 1p4pk — CHANGES_REQUESTED** [review]
See comment before

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java [hunk 1]
```diff
@@ -56,19 +55,23 @@ public void write(final IN in, final Context context) throws IOException {
             } else {
                 log.debug("Adding elements to buffer. Buffer size: {}", this.elements.size());
                 this.elements.add(this.schemaSerializer.serialize(in, context));
+                if (context.timestamp() != null)
+                    this.lastTimestamp = Math.max(this.lastTimestamp, context.timestamp());
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java:59]: add brackets. This leads to really bad mistakes

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java [hunk 2]
```diff
@@ -42,23 +40,30 @@ public InfluxDBCommitter(final InfluxDBConfig config) {
     // This method is called only when a checkpoint is set
     @SneakyThrows
     @Override
-    public List<Void> commit(final List<Void> committables) throws IOException {
+    public List<Long> commit(final List<Long> committables) {
         log.info("A checkpoint is set.");
-        this.writeCheckpointDataPoint();
+        long lastTimestamp = 0;
+        if (committables.size() >= 1) {
+            lastTimestamp = committables.get(committables.size() - 1);
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java:47]: should go to 0 hardcoded right?

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java [hunk 3]
```diff
@@ -42,23 +40,30 @@ public InfluxDBCommitter(final InfluxDBConfig config) {
     // This method is called only when a checkpoint is set
     @SneakyThrows
     @Override
-    public List<Void> commit(final List<Void> committables) throws IOException {
+    public List<Long> commit(final List<Long> committables) {
         log.info("A checkpoint is set.");
-        this.writeCheckpointDataPoint();
+        long lastTimestamp = 0;
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java:45]: Why init with 0? Rather null right? So influx would use its own timestamp and not 0

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBCommittableSerializer.java [hunk 4]
```diff
@@ -33,12 +33,13 @@ public int getVersion() {
...
+    public byte[] serialize(final Long obj) {
         return SimpleVersionedStringSerializer.INSTANCE.serialize(String.valueOf(obj));
     }
 
     @Override
-    public Void deserialize(final int version, final byte[] serialized) throws IOException {
-        return null;
+    public Long deserialize(final int version, final byte[] serialized) throws IOException {
+        return Long.valueOf(
+                SimpleVersionedStringSerializer.INSTANCE.deserialize(version, serialized));
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBCommittableSerializer.java:43]: Reimplement the SimpleVersionedStringSerializer to a SimpleVersionedLongSerializer. ATM, it does not make sense...

**Comment by raminqaf** [comment]
See #22

**Pull request closed without merging**